src/open_samus_returns_rando/files/custom/room_names.lua

```diff
@@ -0,0 +1,91 @@
+Game.DoFile("system/scripts/cc_to_room_name.lc")
+
+RoomNameGui = RoomNameGui or {
+    ui = nil,
+    cameraDict = RANDO_CC_DICTIONARY,
+}
+
+function RoomNameGui.GetRoomName(camera)
+    local dict = RoomNameGui.cameraDict
+    local scenario = Scenario.CurrentScenarioID
+    if dict == nil then
+        Game.LogWarn(0, "No camera dict present!")
+        return
+    end
+
+    local scenario_dict = dict[scenario]
+    if scenario_dict == nil then
+        Game.LogWarn(0, scenario .. " has nil dict!")
+        return nil
+    end
+
+    local rando_name = scenario_dict[camera]
+    return rando_name
+end
+
+function RoomNameGui.Init()
+    if RoomNameGui.ui then
+        GUI.DestroyDisplayObject(RoomNameGui.ui)
+        RoomNameGui.ui = nil
+    end
+
+    local hud = GUI.GetDisplayObject("IngameMenuRoot.IngameMenuComposition.LowerComposition.LowerInfoComposition")
+    local ui = hud:AddChild((GUI.CreateDisplayObjectEx("RoomNameText", "CText", {
+        X = "0.14400",
+        Y = "-0.05900",
+        SizeX = "0.12499",
+        SizeY = "0.05833",
+        ScaleX = "1.00000",
+        ScaleY = "1.00000",
+        Angle = "0.00000",
+        FlipX = false,
+        FlipY = false,
+        ColorR = "0.68000",
+        ColorG = "0.83000",
+        ColorB = "0.93000",
+        ColorA = "1.00000",
+        Enabled = true,
+        Visible = true,
+        SkinItemType = "",
+        Text = "Room Name",
+        Font = "digital_small",
+        TextAlignment = "Center",
+        Autosize = true,
+        Outline = true,
+        EmbeddedSpritesSuffix = "",
+        BlinkColorR = "1.00000",
+        BlinkColorG = "1.00000",
+        BlinkColorB = "1.00000",
+        BlinkAlpha = "1.00000",
+        Blink = "-1.00000"
+    })))
+
+    RoomNameGui.ui = ui
+
+    local current_cc = ""
+
+    RoomNameGui.Update(current_cc)
+end
+
+function RoomNameGui.Update(new_cc)
+    if type(new_cc) ~= "string" then
+        GUI.LogWarn(0, "collision camera is not string")
```

Comment:
Same but also I don't think that this function exists on `GUI` only on `Game` afaik.